packages/runtime/container-runtime/src/opLifecycle/remoteMessageProcessor.ts

```diff
@@ -116,10 +121,20 @@ export class RemoteMessageProcessor {
 
 		// Do a final unpack of runtime messages in case the message was not grouped, compressed, or chunked
 		unpackRuntimeMessage(message);
-		return {
-			messages: [message as InboundSequencedContainerRuntimeMessageOrSystemMessage],
-			batchStartCsn,
-		};
+		this.processorBatch.push(message as InboundSequencedContainerRuntimeMessage);
+		// this.batchStartCsn is undefined only if we have processed all messages in the batch.
+		// If it's still defined, we're still in the middle of a batch, so we return nothing, letting
+		// containerRuntime know that we're waiting for more messages to complete the batch.
+		if (this.batchStartCsn === undefined) {
+			const messages = [...this.processorBatch];
+			this.processorBatch.length = 0;
+			return {
+				messages,
+				batchStartCsn,
+			};
+		}
+		// batch not yet complete
+		return undefined;
```

Comment:
Super nit! Consider inverting the `if (this.batchStartCsn === undefined)` check - then it does the early return of `undefined` if we're in the middle of a batch.  And the couple lines of actual code bump out one indentation level.